Fix logging of key changes to be consistent

barneysowood opened the pull request:
### What does this PR do?

Fixes logging of master aes and session keys changes to be correct and consistent. This was useful when testing issues in #68085 to understand what is actually happening.

Also removes some duplication of code.

### What issues does this PR fix or reference?

Debug logging for master AES and session keys was incomplete and inconsistent across AsyncAuth._authenticate() and SAuth.authenticate()

### Previous Behavior

 AsyncAuth._authenticate() logged both master AES and session key changes as "The master's aes key has changed".
SAuth.authenticate() didn't log anything.

### New Behavior

Both functions consistently log:

* "Got new master aes key." for a new master AES key
* "The master's aes key has changed." for a change in master AES key
* "The master's session key has changed." for a new session key.

The wording isn't ideal, but assuming that people may be used to those messages I didn't want to change too much.

### Commits signed with SSH key?
Yes

<!-- Please review Salt's Contributing Guide for best practices and guidance in
choosing the right branch:
https://docs.saltproject.io/en/master/topics/development/contributing.html -->

<!-- Additional guidance for pull requests can be found here:
https://docs.saltproject.io/en/master/topics/development/pull_requests.html -->

<!-- See GitHub's page on GPG signing for more information about signing commits
with GPG:
https://help.github.com/articles/signing-commits-using-gpg/ -->

barneysowood committed:
Fix logging of key changes to be consistent

Fixes logging of master aes and session keys changes to be correct and
consistent. Also simplifies the code and reduces duplication.

barneysowood commented:
@dwoz - if you aren't up for merging this before 3006.13, no worries - I can always bring it back at a later stage

twangboy reviewed (changes requested):
Could we get a changelog and some tests?

dwoz commented:
> @dwoz - if you aren't up for merging this before 3006.13, no worries - I can always bring it back at a later stage

We staged `3006.13` last night so it didn't make it in. We can merge it when it's ready and it can still go into a release. We're pretty much capable of doing weekly releases now if we need them.

barneysowood committed:
Add tests of log messages for master key rotation

Adds testing of debug log messasges for master key and session key
roatation. Added to existing test_auth_aes_key_rotation and
test_sauth_aes_key_rotation as otherwise I would have to duplicate most
of what they cover.

barneysowood committed:
Add changelog
